src/compositor/seat.rs

```diff
@@ -26,6 +29,49 @@ impl Seat {
                meta: false,
                ..Seat::default() }
     }
+
+    pub fn clear_focus(&mut self) {
+        if let Some(focused_view) = self.focused.take() {
+            focused_view.activate(false);
+        }
+        with_handles!([(seat: {&mut self.seat})] => {
+            seat.keyboard_clear_focus();
+        }).unwrap();
+    }
+
+    pub fn focus_view(&mut self, view: Rc<View>, views: &mut Vec<Rc<View>>) {
+        if let Some(ref focused) = self.focused {
+            if focused == &view {
```

Comment:
(I don't believe this will trigger a move. If it does then don't do this).

`*focused == view` would be clearer